debezium-testing/debezium-testing-system/src/main/java/io/debezium/testing/system/tools/databases/mongodb/OcpMongoShardedController.java

```diff
@@ -156,30 +158,32 @@ private void uploadAndExecuteMongoScript(String scriptLocation, MongoComponents
 
         podResource.file(containerPath)
                 .upload(scriptPath);
-        executeCommadOnComponent(component,
+        executeCommandOnComponent(component,
                 "mongosh", "localhost:" + component.getPort(), "-f", containerPath);
     }
 
     public enum MongoComponents {
-        CONFIG("mongo-config", 27019, List.of("mongosh", "localhost:27019", "--eval",
+        CONFIG("mongo-config", 27019, new String[]{ "mongosh", "localhost:27019", "--eval",
                 "rs.initiate({ _id: \"cfgrs\", configsvr: true, members: [{ _id : 0, host : \"mongo-config." + ConfigProperties.OCP_PROJECT_MONGO
-                        + ".svc.cluster.local:27019\" }]})")),
-        SHARD1R1("mongo-shard1r1", 27018, List.of("mongosh", "localhost:27018", "--eval",
+                        + ".svc.cluster.local:27019\" }]})" }),
+        SHARD1R1("mongo-shard1r1", 27018, new String[]{ "mongosh", "localhost:27018", "--eval",
                 "rs.initiate({_id: \"shard1rs\", members: [{ _id : 0, host : \"mongo-shard1r1." + ConfigProperties.OCP_PROJECT_MONGO
-                        + ".svc.cluster.local:27018\" }]})")),
-        SHARD2R1("mongo-shard2r1", 27018, List.of("mongosh", "localhost:27018", "--eval",
+                        + ".svc.cluster.local:27018\" }, { _id : 1, host : \"mongo-shard1r2." + ConfigProperties.OCP_PROJECT_MONGO
+                        + ".svc.cluster.local:27018\" }]})" }),
+        SHARD1R2("mongo-shard1r2", 27018, null),
+        SHARD2R1("mongo-shard2r1", 27018, new String[]{ "mongosh", "localhost:27018", "--eval",
                 "rs.initiate({_id: \"shard2rs\", members: [{ _id : 0, host : \"mongo-shard2r1." + ConfigProperties.OCP_PROJECT_MONGO
-                        + ".svc.cluster.local:27018\" }]})")),
-        SHARD3R1("mongo-shard3r1", 27018, List.of("mongosh", "localhost:27018", "--eval",
+                        + ".svc.cluster.local:27018\" }]})" }),
+        SHARD3R1("mongo-shard3r1", 27018, new String[]{ "mongosh", "localhost:27018", "--eval",
                 "rs.initiate({_id: \"shard3rs\", members: [{ _id : 0, host : \"mongo-shard3r1." + ConfigProperties.OCP_PROJECT_MONGO
-                        + ".svc.cluster.local:27018\" }]})")),
+                        + ".svc.cluster.local:27018\" }]})" }),
         MONGOS("mongo-mongos", 27017, null);
 
         private final String name;
         private final int port;
-        private final List<String> initCommand;
+        private final String[] initCommand;
```

Comment:
What was the thinking behind changing the list to array?

Reply:
It is only used as array and passed to executeCommand, so it was always converted anyway